app/assets/javascripts/leaflet.share.js

```diff
@@ -411,34 +416,20 @@ L.OSM.share = function (options) {
       $("#mapnik_image_width").text(mapWidth);
       $("#mapnik_image_height").text(mapHeight);
 
-      const layerId = map.getMapBaseLayerId();
-      const layerKeys = new Map([
-        ["mapnik", "standard"],
-        ["cyclemap", "cycle_map"],
-        ["transportmap", "transport_map"]
-      ]);
+      const canDownloadImage = layer && layer.options.canDownloadImage;
```

Comment:
This doesn't guarantee a boolean valued result - specifically if the `canDownloadImage` key is missing then the result is `undefined` not `false` and at least in Firefox calling `toggle(undefined)` toggles the current visibility instead of always hiding the element though I don't think the behaviour is defined at all by jQuery.

Reply:
```suggestion
      const canDownloadImage = !!(layer && layer.options.canDownloadImage);
```
or
```suggestion
      const canDownloadImage = Boolean(layer && layer.options.canDownloadImage);
```
[Modern js](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) would be (available across browsers since September 2020):
```suggestion
      const canDownloadImage = layer?.options.canDownloadImage ?? false;
```

Reply:
Updated to `Boolean( ... )`, eslint doesn't like `!!` too.